services/comprehensive_blog_generator.py

```
@@ -0,0 +1,730 @@
+            if hasattr(obj, 'isoformat'):  # datetime objects
+                return obj.isoformat()
+            elif hasattr(obj, '__dict__'):  # custom objects
+                return str(obj)
+            else:
+                return str(obj)
+        
+        def safe_json_dumps(data, indent=2):
+            """Safely serialize data to JSON with fallback handling."""
+            try:
+                return json.dumps(data, indent=indent, ensure_ascii=False, default=safe_json_serializer)
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)